src/simple_commander/main.py

```diff
@@ -401,14 +428,25 @@ def start(self, socket, name, *args, **kwargs):
             'units': self.get_units(),
             'frequency': STEP_INTERVAL}}
         socket.send_str(json.dumps(start_conditions))
-        return my_hero
 
-    def add_bonus(self, bullet):
+    def add_hits(self, bullet):
         for unit in self.units:
```

Comment:
self.unit is a dictionary, so instead of all units iteration why don't you extract the unit by ID?